openwisp_network_topology/templates/netjsongraph/netjsongraph-script.html

```diff
@@ -165,6 +165,38 @@
             }
         }
 
+        fetch(url, {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+                Accept: "application/json",
+            },
```

Comment:
Do we really need to use an additional fetch request here? Technically we should get the number of nodes using just `graph.data.nodes.length` without the use of any additional fetch.

Reply:
Vaishnav is right here @pandafy, this will request the same URL twice.

You have to do this request before `new NetJSONGraph(url...` and then pass the NetJSON object instead of the URL.
This mean also that instead of using `graph.setConfig()` we can simply construct the config conditionally and pass the result during initialization, without resetting it, which is what I meant in the issue description but reading it again now I see that I didn't make this point explicit enough.